cli/internal/services/authentication/authentication.go

```diff
@@ -1,35 +1,87 @@
 package authentication
 
 import (
+	"cli/internal/services/utils"
 	"encoding/json"
+	"fmt"
+	"net/http"
 	"os"
 	"path/filepath"
+	"sync"
 )
 
-const (
-	fileName = "session"
-)
-
+// Authentication tokens.
 type Authentication struct {
 	SessionToken string `json:"access_token,omitempty"`
 	RefreshToken string `json:"refresh_token,omitempty"`
 }
 
+const (
+	// The name of the file where the session and refresh token are stored.
+	fileName = "session"
+)
+
+var (
+	// The Keycloak Configuration
+	// TODO: Read this from a config file or environment variables
+	keycloakConfig = KeycloakConfig{
+		URL: "http://localhost:8080",
+		Realm: "ihp-realm",
+		ClientId: "ihp-cli",
+	}
+
+	// The Embedded Server Configuration
+	// TODO: Read this from a config file or environment variables
+	embeddedServerConfig = utils.EmbeddedServerConfig{
+		Port: 3000,
```

Comment:
I do not think giving the user a configuration option for this port or hardcoding it is useful. I think choosing a port within a given range based on whether or not it is already in use may be the way to go.